modules/jabba/1.0/jabba.smk

```diff
@@ -0,0 +1,349 @@
+
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _jabba_install_fragcounter,
+    _jabba_install_dryclean,
+    _jabba_install_jabba,
+    _jabba_input_bam,
+    _jabba_input_junc_bedpe,
+    _jabba_input_junc_vcf,
+    _jabba_link_normal_rds,
+    _jabba_link_dryclean_normal_rds,
+    _jabba_link_graph_rds,
+    _jabba_all
+
+
+##### RULES #####
+
+### Adjust coverage for GC and mappability ###
+
+# Install JaBba suite from github
+rule _jabba_install_fragcounter:
+    output:
+        complete = CFG["dirs"]["fragcounter"] + "fragcounter.installed"
+    conda: CFG["conda_envs"]["jabba"]
+    shell:
+        op.as_one_line("""
+        Rscript -e 'Sys.setenv(R_REMOTES_NO_ERRORS_FROM_WARNINGS = TRUE)'
+                -e 'remotes::install_github("mskilab/fragCounter", upgrade = FALSE)' 
+            &&
+        touch {output.complete}
+        """)
+
+rule _jabba_install_dryclean:
+    output:
+        complete = CFG["dirs"]["dryclean"] + "dryclean.installed"
+    conda: CFG["conda_envs"]["jabba"]
+    shell:
+        op.as_one_line("""
+        Rscript -e 'Sys.setenv(R_REMOTES_NO_ERRORS_FROM_WARNINGS = TRUE)' 
+                -e 'remotes::install_github("mskilab/dryclean", upgrade = FALSE)'
+            &&
+        touch {output.complete}
+        """)
+
+rule _jabba_install_jabba:
+    output:
+        complete = CFG["dirs"]["jabba"] + "jabba.installed"
+    conda: CFG["conda_envs"]["jabba"]
+    params:
+        cplex_dir = CFG["CPLEX_DIR"]
+    shell:
+        op.as_one_line(""" 
+        Rscript -e 'Sys.setenv(R_REMOTES_NO_ERRORS_FROM_WARNINGS = TRUE)'
+                -e 'Sys.setenv(CPLEX_DIR = "{params.cplex_dir}")'
+                -e 'remotes::install_github("mskilab/JaBba", upgrade = FALSE)'
+            &&
+        touch {output.complete}
+        """)
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _jabba_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai",
+        crai = CFG["dirs"]["inputs"]+ "bam/{seq_type}--{genome_build}/{sample_id}.crai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+        op.relative_symlink(input.bai, output.crai)
+
+rule _jabba_input_junc_bedpe:
+    input:
+        junc = CFG["inputs"]["sample_junc"]
+    output:
+        junc = CFG["dirs"]["inputs"] + "junc/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}.bedpe"
+    run:
+        op.relative_symlink(input.junc, output.junc)
+
+rule _jabba_input_junc_vcf:
+    input:
+        junc = CFG["inputs"]["sample_junc"]
+    output:
+        junc = CFG["dirs"]["inputs"] + "junc/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}.vcf"
```

Comment:
Does it work with compressed vcf files or do you need to decompress them? If the input is gzipped, this link might be misleading